openquake/hazardlib/shakemap/parsers.py

```diff
@@ -441,64 +455,64 @@ def download_station_data_file(usgs_id, save_to_home=False):
-            logging.info('Downloading stationlist.json')
-            stations_json_str = urlopen(stationlist_url).read()
-            try:
-                stations = read_usgs_stations_json(stations_json_str)
-            except (LookupError, UnicodeDecodeError, JSONDecodeError) as exc:
-                logging.info(str(exc))
-                raise
-            original_len = len(stations)
-            try:
-                seismic_len = len(
-                    stations[stations['station_type'] == 'seismic'])
-            except KeyError:
-                msg = (f'{original_len} stations were found, but the'
-                       f' "station_type" is not specified, so we can not'
-                       f' identify the "seismic" stations.')
-                logging.info(msg)
-                raise LookupError(msg)
-            df = usgs_to_ecd_format(stations, exclude_imts=('SA(3.0)',))
-            if save_to_home:
-                homedir = os.path.expanduser('~')
-                stations_usgs = os.path.join(homedir, 'stations_usgs.csv')
-                stations.to_csv(stations_usgs, index=False)
-                stations_oq = os.path.join(homedir, 'stations_oq.csv')
-                df.to_csv(stations_oq, index=False)
-            if len(df) < 1:
-                if original_len > 1:
-                    if seismic_len > 1:
-                        msg = (f'{original_len} stations were found, but the'
-                               f' {seismic_len} seismic stations were all'
-                               f' discarded')
-                        logging.info(msg)
-                        raise LookupError(msg)
-                    else:
-                        msg = (f'{original_len} stations were found, but none'
-                               f' of them are seismic')
-                        logging.info(msg)
-                        raise LookupError(msg)
+    shakemap = get_preferred_shakemap(shakemaps)
+    contents = shakemap['contents']
+    if 'download/stationlist.json' in contents:
+        stationlist_url = contents.get('download/stationlist.json')['url']
+        logging.info('Downloading stationlist.json')
+        stations_json_str = urlopen(stationlist_url).read()
+        try:
+            stations = read_usgs_stations_json(stations_json_str)
+        except (LookupError, UnicodeDecodeError, JSONDecodeError) as exc:
+            logging.info(str(exc))
+            raise
+        original_len = len(stations)
+        try:
+            seismic_len = len(
+                stations[stations['station_type'] == 'seismic'])
+        except KeyError:
+            msg = (f'{original_len} stations were found, but the'
+                   f' "station_type" is not specified, so we can not'
+                   f' identify the "seismic" stations.')
+            logging.info(msg)
+            raise LookupError(msg)
+        df = usgs_to_ecd_format(stations, exclude_imts=('SA(3.0)',))
+        if save_to_home:
```

Comment:
Why do we want to save in the home directory?

Reply:
It is explained in the docstring. I wanted to have a simple way to check the contents of the station data before and after the conversion, for debugging purposes. It is disabled by default.